cli/src/diff_util.rs

```diff
@@ -1110,7 +1110,11 @@ pub fn show_diff_summary(
     path_converter: &RepoPathUiConverter,
 ) -> io::Result<()> {
     async {
-        while let Some((repo_path, diff)) = tree_diff.next().await {
+        while let Some(TreeDiffEntry {
+            target: repo_path,
+            value: diff,
```

Comment:
I really don't like `diff` as the name for the view because it fundamentally is a pair of values (well `MaterializedTreeValue`).  I would rather rename it to `values` or `tree_values` and use that locally.  I mean, my strongest preference is to split it into `source_value` and `target_value`, but I was not able to accomplish that refactor despite three attempts

Reply:
I actually think `diff` is a good name for a pair of values :) But I won't insist, and maybe that refactoring you mention  will happen eventually anyway.